File: backend/app/schemas/look.py
"""Schemas Pydantic para Look e LookItem."""
from pydantic import BaseModel, Field
from datetime import datetime
from typing import Optional, List
from decimal import Decimal


# ── LookItem ─────────────────────────────────────────────────────────────────

class LookItemCreate(BaseModel):
    product_id: int = Field(..., gt=0)
    variant_id: Optional[int] = None
    position: int = Field(0, ge=0)


class LookItemResponse(BaseModel):
    id: int
    look_id: int
    product_id: int
    variant_id: Optional[int] = None
    position: int
    product_name: Optional[str] = None
    variant_description: Optional[str] = None
    product_image_url: Optional[str] = None
    unit_price: Optional[float] = None
    created_at: datetime

    model_config = {"from_attributes": True}


# ── Look ─────────────────────────────────────────────────────────────────────

class LookCreate(BaseModel):
    name: str = Field(..., min_length=1, max_length=200)
    description: Optional[str] = None
    customer_id: Optional[int] = None
    is_public: bool = True
    discount_percentage: float = Field(0.0, ge=0, le=100)
    items: List[LookItemCreate] = []


class LookUpdate(BaseModel):
    name: Optional[str] = Field(None, min_length=1, max_length=200)
    description: Optional[str] = None
    is_public: Optional[bool] = None
    discount_percentage: Optional[float] = Field(None, ge=0, le=100)


class LookResponse(BaseModel):
    id: int
    tenant_id: Optional[int] = None
    name: str
    description: Optional[str] = None
    customer_id: Optional[int] = None
    is_public: bool
    discount_percentage: float
    items: List[LookItemResponse] = []
    total_price: float = 0.0
    items_count: int = 0
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class LookListResponse(BaseModel):
    id: int
    name: str
    description: Optional[str] = None
    customer_id: Optional[int] = None
    is_public: bool
    discount_percentage: float
    items_count: int = 0
    total_price: float = 0.0
    created_at: datetime

    model_config = {"from_attributes": True}

File: mobile/types/look.ts
// Types para Lookbook e Wishlist

export interface LookItemResponse {
  id: number;
  look_id: number;
  product_id: number;
  variant_id: number | null;
  position: number;
  product_name: string | null;
  variant_description: string | null;
  product_image_url: string | null;
  unit_price: number | null;
  created_at: string;
}

export interface Look {
  id: number;
  tenant_id: number | null;
  name: string;
  description: string | null;
  customer_id: number | null;
  is_public: boolean;
  discount_percentage: number;
  items: LookItemResponse[];
  total_price: number;
  items_count: number;
  created_at: string;
  updated_at: string;
}

export interface LookList {
  id: number;
  name: string;
  description: string | null;
  customer_id: number | null;
  is_public: boolean;
  discount_percentage: number;
  items_count: number;
  total_price: number;
  created_at: string;
}

export interface CreateLookDTO {
  name: string;
  description?: string;
  customer_id?: number;
  is_public?: boolean;
  discount_percentage?: number;
  items: {
    product_id: number;
    variant_id?: number;
    position?: number;
  }[];
}

export interface WishlistItem {
  id: number;
  customer_id: number;
  product_id: number;
  variant_id: number | null;
  look_id: number | null;
  notified: boolean;
  notified_at: string | null;
  notes: string | null;
  product_name: string | null;
  variant_description: string | null;
  customer_name: string | null;
  product_image_url: string | null;
  in_stock: boolean;
  created_at: string;
}

export interface DemandItem {
  product_id: number;
  product_name: string;
  variant_id: number | null;
  variant_description: string | null;
  waiting_count: number;
  potential_revenue: number;
  product_image_url: string | null;
}
